### PR TITLE
feat: add tri-state indeterminate checkbox feature

### PR DESCRIPTION
**Pull request by jorenbroekema**
[See WAI-ARIA docs on this pattern](https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-2/checkbox-2.html)

- [x] Add test for tri-state feature
- [x] Implement feature

GIF

### CODE REVIEW

**Comment by gerjanvangeest** [comment]
In the wai-aria example, when all/no child boxes are checked the indeterminate only toggles between checked and unchecked. In your case it remembers the indeterminate state before.

Might be a good addition.

File: packages/checkbox-group/test/lion-checkbox-indeterminate.test.js [hunk 1]
```diff
@@ -428,4 +429,79 @@ describe('<lion-checkbox-indeterminate>', () => {
     expect(elIndeterminate?.hasAttribute('indeterminate')).to.be.false;
     expect(elIndeterminate?.checked).to.be.true;
   });
+
+  // https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-2/checkbox-2.html
+  describe('Tri-state', () => {
```
Comment (gerjanvangeest) [inline packages/checkbox-group/test/lion-checkbox-indeterminate.test.js:434]:
```suggestion
  describe('mixed-state', () => {
  it('can have a mixed-state (using mixed-state attribute), none -> indeterminate -> all, cycling through', async () => {
```

**Comment by jorenbroekema** [comment]
Oooh yeah you're right, I think that's a bit awkward, the WAI ARIA one situation makes more sense indeed. Will change it and add a test for that case tomorrow, shouldn't be too difficult :)

File: packages/checkbox-group/test/lion-checkbox-indeterminate.test.js [hunk 2]
```diff
@@ -428,4 +429,80 @@ describe('<lion-checkbox-indeterminate>', () => {
     expect(elIndeterminate?.hasAttribute('indeterminate')).to.be.false;
     expect(elIndeterminate?.checked).to.be.true;
   });
+
+  // https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-2/checkbox-2.html
+  describe('mixed-state', () => {
+  it('can have a mixed-state (using mixed-state attribute), none -> indeterminate -> all, cycling through', async () => {
+    it('can have a tri-state (using tri-state attribute), none -> indeterminate -> all, cycling through', async () => {
```
Comment (gerjanvangeest) [inline packages/checkbox-group/test/lion-checkbox-indeterminate.test.js:436]: Now there is a double "it"

```suggestion
```

Comment (jorenbroekema) [reply packages/checkbox-group/test/lion-checkbox-indeterminate.test.js:436]: Giving me bad suggestions to make CI fail, I seeeee hehe

**Comment by jorenbroekema** [comment]
I was super hungover yesterday so I couldn't get to it, trainings today and tomorrow, so hopefully this weekend I can get to it and finish it

**Comment by jorenbroekema** [comment]
Done :) was actually slightly harder to implement than I thought initially but it works great now, also for nested indeterminate groups with mixed state 🎉

**Comment by gerjanvangeest** [comment]
The WAI aria example and your implementation are different:

**Comment by jorenbroekema** [comment]
Huh, weird, I can't reproduce your first gif @gerjanvangeest , for me it works (toggling between 3 states like in WAI ARIA)

**Comment by gerjanvangeest** [comment]
> Huh, weird, I can't reproduce your first gif @gerjanvangeest , for me it works (toggling between 3 states like in WAI ARIA)

Ok, let me double check why i have it.

**Comment by gerjanvangeest** [comment]
Oke, my bad. I was looking at the wrong example 🙈

**Review by gerjanvangeest — APPROVED** [review]

**Pull request merged**